cli/azd/pkg/environment/environment.go

```diff
@@ -256,3 +230,114 @@ func (e *Environment) Environ() []string {
 
 	return envVars
 }
+
+func (e *Environment) watchForChanges() error {
+	if e.watcher != nil {
+		return nil
+	}
+
+	// Don't setup the watcher if the .env file doesn't exist yet
+	envFilePath := filepath.Join(e.Root, azdcontext.DotEnvFileName)
+	if _, err := os.Stat(envFilePath); errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return fmt.Errorf("failed creating file watcher, %w", err)
+	}
+
+	if err := watcher.Add(envFilePath); err != nil {
+		return fmt.Errorf("failed watching environment file '%s', %w", envFilePath, err)
+	}
+
+	e.watcher = watcher
```

Comment:
Consider adding a mutex to avoid multiple a goroutine leaked from concurrent `Save` calls.